syntax = "proto3";

package NAlice;

option go_package = "a.yandex-team.ru/alice/megamind/protos/common";
option java_package = "ru.yandex.alice.megamind.protos.common";

import "mapreduce/yt/interface/protos/extension.proto";

// These parameters identify the way of user interatction
message TAnalyticsTrackingModule {
    option (NYT.default_field_flags) = SERIALIZATION_YT;

    enum EOrigin {
        Undefined = 0;
        Web = 1;
        Scenario = 2;
        SmartSpeaker = 3 [deprecated = true];
        RemoteControl = 4;
        Proactivity = 5;
        Cast = 6;
        Timetable = 7;
        Push = 8;
        SearchApp = 9 [deprecated = true];
        ThisClient = 10;
        SmartTv = 11;
    }

    string ProductScenario = 1 [json_name = "product_scenario", (NYT.column_name) = "product_scenario"];

    // The origin of the request/event, like website or remote control
    EOrigin Origin = 2 [json_name = "origin", (NYT.column_name) = "origin"];

    // The purpose identifies a user's intent during the action
    // Why the request was sent and what is expected in the response
    // Slug like get_weather, next_track, and so on
    string Purpose = 3 [json_name = "purpose", (NYT.column_name) = "purpose"];

    // Any extra information about the origin, like id of marketing campaigns
    string OriginInfo = 4 [json_name = "origin_info", (NYT.column_name) = "origin_info"];
}
